src/main/java/domain/card/Deck.java

```diff
@@ -1,25 +1,38 @@
-package domain;
+package domain.card;
 
-import domain.strategy.DrawStrategy;
+import domain.card.strategy.DrawStrategy;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.Deque;
 import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class DeckGenerator {
+public class Deck {
 
-    public static Deck generateDeck(DrawStrategy drawStrategy) {
+    private final DrawStrategy drawStrategy;
+    private final Deque<TrumpCard> deck;
+
+    public Deck(DrawStrategy drawStrategy) {
+        this.deck = generateDeck();
+        this.drawStrategy = drawStrategy;
+    }
+
+    public TrumpCard drawCard() {
+        return drawStrategy.draw(deck);
+    }
+
+    private LinkedList<TrumpCard> generateDeck() {
         List<Suit> suits = Arrays.stream(Suit.values()).toList();
         Set<TrumpCard> trumpCards = new HashSet<>();
         for (Suit suit : suits) {
```

Comment:
Enum의 values() 로도 iteration이 가능합니다!

```suggestion
        for (Suit suit : Suit.values()) {
```

Reply:
`values()`를 사용하지 않고 따로 Stream을 이용해서 `Suit` 리스트를 생성하고 있었네요. `values()`를 이용하여 List를 따로 생성하지 않아도 될 것 같아요. 피드백 감사합니다 ☺️